src/raw/unix.rs

```diff
@@ -414,6 +418,32 @@ impl Socket {
     }
 
 
+    ///Sets whether this socket will be inherited to child porcesses or not.
+    pub fn set_inheritable(&self, value: bool) -> io::Result<()> {
+        // Some (or possibly all?) OS's support the `FIOCLEX` and `FIONCLEX`

```

Comment:
The flags aren't even defined in `libc` crate and there is no such thing as `#if defined(libc::FIOCLEX)` in Rust, so unless somebody (and right not it's not going to be me :wink:) does the work to figuring out which platforms support these `ioctl`s and submits these to the `libc` I'll keep things as is. (I suspect we'll need this portable fallback anyways because some platform, possibly macOS, will not support the `ioctl` way…)